src/components/forms/InputField.tsx

```diff
@@ -78,86 +106,129 @@ export function InputField(props: Props) {
       )}
 
       <div className="relative">
-        <DebounceInput
-          style={{
-            backgroundColor: colors.$1,
-            borderColor: colors.$5,
-            color: colors.$3,
-            ...props.style,
-          }}
-          min={props.min}
-          max={props.type === 'date' ? '9999-12-31' : undefined}
-          maxLength={props.maxLength}
-          autoComplete={props.autoComplete || 'new-password'}
-          disabled={props.disabled}
-          element={props.element || 'input'}
-          inputRef={props.innerRef}
-          debounceTimeout={props.debounceTimeout ?? 300}
-          required={props.required}
-          id={props.id}
-          type={inputType}
-          className={classNames(
-            `w-full py-2 px-3 rounded text-sm disabled:opacity-75 disabled:cursor-not-allowed ${props.className}`,
-            {
-              'border border-gray-300': props.border !== false,
-            }
-          )}
-          placeholder={props.placeholder || ''}
-          onBlur={(event) => {
-            event.target.value =
-              event.target.value === '' && props.type === 'number'
-                ? '0'
-                : event.target.value;
-
-            props.onValueChange && props.onValueChange(event.target.value);
-            props.onChange && props.onChange(event);
-          }}
-          onChange={(event) => {
-            if (
-              props.element === 'textarea' &&
-              reactSettings.preferences.auto_expand_product_table_notes
-            ) {
-              const scrollHeight = event.target.scrollHeight + 2;
-
-              if (scrollHeight < 200) {
-                event.target.style.height = scrollHeight + 'px';
+        {props.type === 'number' ? (
```

Comment:
This is getting a bit noisy. Is it an easy win to split the components? Eg. NumberInputField?

Reply:
> This is getting a bit noisy. Is it an easy win to split the components? Eg. NumberInputField?

@beganovich Sure thing, I also think it's best to change it now and make maintenance easier for us. I've made the necessary changes, and you can take a look to see if you agree with them. However, I'll need some time to test these significant changes in our app to avoid any edge case bugs. I'll return the PR to "ready for review" once I finish the full testing.